app/app/questions/page.jsx

```diff
@@ -0,0 +1,84 @@
+'use client';
+
+import { useState, useEffect } from 'react';
+import QuestionCard from '../../components/Questioncard';
+import PreviousNext from '../../components/Prevnext';
+
+function Exam() {
+  const [questions, setQuestions] = useState([]);
+  const [currentQuestionIndex, setCurrentQuestionIndex] = useState(0);
+  const [userAnswers, setUserAnswers] = useState({});
+
+  useEffect(() => {
+    const fetchQuestions = async () => {
+      try {
+        const apiUrl = '/api/questions';
+        const response = await fetch(apiUrl);
+        if (!response.ok) {
+          throw new Error('Failed to fetch questions');
+        }
+        const data = await response.json();
+        setQuestions(data);
+      } catch (error) {
+        console.error('Error fetching questions:', error);
+      }
+    };
+
+    fetchQuestions();
+  }, []);
+
+  const handleAnswer = (answer) => {
+    const updatedAnswers = { ...userAnswers, [currentQuestionIndex]: answer };
+    setUserAnswers(updatedAnswers);
+    if (currentQuestionIndex < questions.length - 1) {
+      setCurrentQuestionIndex(currentQuestionIndex + 1);
+    }
+  };
+
+  const handleNext = () => {
+    if (currentQuestionIndex < questions.length - 1) {
+      setCurrentQuestionIndex(currentQuestionIndex + 1);
+    }
+  };
+
+  const handlePrevious = () => {
+    if (currentQuestionIndex > 0) {
+      setCurrentQuestionIndex(currentQuestionIndex - 1);
+    }
+  };
+
+  if (questions.length === 0) {
+    return <div>Loading...</div>;
+  }
+
+  const currentQuestion = questions[currentQuestionIndex];
+  const answers = currentQuestion.answers;
+
+  return (
+    <div className="flex min-h-screen items-center justify-center bg-white ">
+      <div className="mx-auto w-full max-w-4xl rounded-xl p-10">
+        <QuestionCard
+          question={currentQuestion.question}
+          answers={answers}
+          onAnswer={handleAnswer}
+        />
+
+        <div>
+          <PreviousNext
+            showPrevious={currentQuestionIndex > 0}
+            showNext={currentQuestionIndex < questions.length}
+            onPrevious={handlePrevious}
```

Comment:
<img width="773" alt="image" src="https://github.com/user-attachments/assets/3c0ae9f5-762c-42da-9410-006c559ac8c0" />

Also "Next" button should probably always be from the right part (also can be fixed later in another task)